src/common/hist_util.cc

```diff
@@ -29,6 +30,34 @@
 namespace xgboost {
 namespace common {
 
+template<typename BinIdxType>
+void GHistIndexMatrix::SetIndexData(BinIdxType* const index_data, size_t batch_threads,
+                                    const SparsePage& batch, size_t rbegin,
+                                    const uint32_t* disps, size_t nbins) {
+  #pragma omp parallel for num_threads(batch_threads) schedule(static)
```

Comment:
Do you have some reasons to use custom amount of threads - ```batch_threads``` ?

Reply:
`batch_threads` is the same as before.